<details>
  <summary><font size="3" color="gold"><b>  HDDA Assignment 1 </b></font></summary>

<details>
  <summary><font size="3" color="gold"><b>James Wandiya </b></font></summary>

<span style="color: gold">Answer 2</span>

<a name="1"> # Packages </a>



In [1]:
import numpy as np
from scipy import stats

<span style="color: gold">Data</span>

In [2]:
x = np.array([126, 700, 275, 500, 188, 510, 0, 255, 502, 280, 387, 229, 347, 310, 90, 95, 102, 300, 572, 257, 0, 167, 0, 10, 230, 502, 173, 126, 209, 235, 315, 150, 433, 391, 140, 426, 0, 270, 300, 651, 453, 0, 300, 392, 250, 109, 110, 40, 560, 345, 427, 280, 52, 323, 130, 40, 230, 250, 569, 365, 385, 287, 300, 430, 110, 60, 195, 0, 420, 130, 550, 332, 132, 0, 417, 218, 319, 150, 220, 275, 1278, 64, 450, 350, 348, 370, 200, 250, 426, 384, 70, 86, 142, 256, 0, 300, 0, 255, 458, 245, 280, 40, 0, 150, 865, 307, 579, 0, 0, 204, 420, 250, 250, 300, 242, 25, 350, 580, 388, 318, 418, 115, 181, 0, 447, 289, 563, 70, 422, 130, 0, 535, 300, 281, 202, 0, 1200, 18, 331, 0, 210, 79, 250, 0, 426, 372, 450, 277, 0, 137, 90, 200, 170, 66, 0, 0, 0, 20, 120, 203, 200, 299, 412, 700, 200, 443, 0, 0, 522, 0, 80, 344, 250, 400, 196, 180, 0, 500, 400, 0, 580, 0, 0, 612, 972, 300, 300, 443, 399, 500, 0, 0, 298, 216, 310, 252, 250, 0, 698, 250, 225, 295, 418, 332, 300, 849, 295, 0, 311, 40, 463, 252, 80, 400, 0, 266, 100, 0, 267, 214, 1000, 172, 0, 2, 0, 211, 228, 350, 0, 51, 217, 0, 363, 286, 310, 0, 386, 400, 0, 242, 218, 495, 74, 391, 580, 318, 174, 838, 100, 233, 0, 0, 0, 30, 300, 0, 150, 660, 477, 480, 378, 51, 800, 0, 183, 200, 253, 380, 216, 294, 255, 347, 290, 270, 263, 606, 280, 1151, 260, 139, 490, 160, 143, 324, 290, 50, 0, 160, 500, 500, 280, 400, 45, 0, 56, 131, 240, 279, 231, 600, 600, 70, 386, 234, 200, 334, 5000, 0, 400, 360, 219, 215, 250, 75, 280, 458, 270, 100, 360, 185, 650, 250, 180, 0, 160, 210, 215, 700, 120, 376, 735, 445, 300, 400, 0, 461, 356, 500, 500, 0, 227, 549, 742, 700, 407, 250, 346, 346, 661, 0, 492, 250, 54, 559, 564, 300, 613, 130, 234, 674, 245, 66, 50, 0, 360, 344, 218, 312, 0])

y = np.array([161, 380, 207, 900, 160, 530, 0, 237, 324, 298, 336, 292, 112, 324, 141, 78, 81, 300, 406, 83, 0, 188, 0, 2, 315, 705, 218, 327, 344, 104, 295, 201, 349, 1231, 120, 165, 0, 1537, 443, 357, 263, 0, 464, 355, 850, 227, 97, 35, 529, 404, 384, 360, 40, 809, 100, 40, 285, 100, 548, 379, 600, 367, 500, 430, 285, 100, 122, 0, 400, 112, 350, 603, 153, 0, 425, 365, 519, 150, 476, 285, 287, 59, 550, 450, 457, 193, 200, 150, 444, 316, 170, 117, 307, 269, 0, 510, 0, 242, 292, 68, 300, 40, 0, 200, 970, 301, 416, 0, 0, 267, 580, 200, 200, 270, 321, 52, 160, 710, 216, 534, 700, 129, 202, 0, 208, 487, 569, 50, 478, 76, 0, 257, 160, 320, 169, 0, 200, 0, 242, 0, 186, 130, 430, 0, 250, 454, 509, 321, 0, 112, 80, 200, 133, 49, 0, 0, 0, 20, 150, 198, 410, 289, 375, 700, 190, 155, 0, 0, 706, 0, 80, 330, 270, 400, 136, 200, 0, 500, 1300, 0, 710, 0, 0, 565, 268, 10, 400, 485, 238, 200, 0, 0, 217, 224, 328, 372, 300, 0, 600, 175, 325, 300, 1209, 453, 485, 578, 224, 0, 325, 40, 150, 120, 90, 290, 0, 337, 100, 0, 121, 207, 1136, 5800, 0, 154, 0, 860, 210, 353, 0, 22, 293, 0, 542, 102, 167, 0, 413, 100, 0, 343, 168, 542, 76, 348, 615, 284, 174, 1115, 400, 148, 0, 0, 0, 40, 711, 0, 110, 390, 310, 598, 325, 0, 270, 0, 241, 260, 255, 944, 212, 301, 122, 231, 367, 430, 674, 505, 290, 552, 200, 97, 570, 120, 128, 574, 525, 80, 0, 362, 400, 1200, 390, 450, 51, 0, 100, 257, 380, 209, 269, 900, 1330, 130, 532, 231, 120, 383, 5000, 0, 600, 280, 242, 126, 300, 55, 329, 653, 218, 200, 235, 220, 1000, 100, 200, 0, 240, 180, 215, 351, 196, 383, 479, 397, 500, 332, 0, 482, 313, 500, 500, 0, 200, 426, 826, 800, 472, 280, 652, 163, 713, 0, 502, 300, 38, 545, 246, 183, 370, 50, 159, 596, 387, 59, 70, 0, 360, 344, 218, 312, 0])

<details>
  <summary><font size="3" color="gold"><b>To Calculate correlation coefficient</b></font></summary>

In [5]:
r = np.corrcoef(x, y)[0, 1]
print(f"Correlation coefficient: r = {r:.4f}")

Correlation coefficient: r = 0.6411


<details>
  <summary><font size="3" color="gold"><b>To Calculate regression equation</b></font></summary>

In [14]:
slope, intercept, r_value, p_value, std_err = stats.linregress(x, y)

print(f"   ŷ = {intercept:.2f} + {slope:.4f}x")
print(f"   or to write like this: Followers = {intercept:.2f} + {slope:.4f} × (People Followed)")

   ŷ = 60.31 + 0.8936x
   or to write like this: Followers = 60.31 + 0.8936 × (People Followed)


<details>
  <summary><font size="3" color="gold"><b>To Forecast Instagram followers of an AIMS student who follows 200
people.</b></font></summary>

In [9]:

y_pred = intercept + slope * 200
print(f"Prediction for 200: {y_pred:.0f}")

Prediction for 200: 239


<span style="color: gold">Insight derived:</span> The model shows that while following more people is associated with having more followers, the relationship is only moderately strong.

<span style="color: gold">Answer 3</span>

<span style="color: gold">Data</span>

In [21]:
#Independent variable (# NbOfContacts)
#Dependent variable (# Sales)
x = np.array([14, 12, 20, 16, 46, 23, 48, 50, 55, 50])
y = np.array([24, 14, 28, 30, 80, 30, 90, 85, 120, 110])

<details>
  <summary><font size="3" color="gold"><b> (a) Determine the regression equation. </b></font></summary>

In [22]:
slope, intercept, r, p, std_err = stats.linregress(x, y)

In [23]:
print(f"   Sales = {intercept:.3f} + {slope:.3f} × Contacts")
print(f"   or: ŷ = {intercept:.3f} + {slope:.3f}x")

   Sales = -12.201 + 2.195 × Contacts
   or: ŷ = -12.201 + 2.195x


<details>
  <summary><font size="3" color="gold"><b> (b)Estimated sales for 40 contacts </b></font></summary>

In [24]:
contacts = 40
predicted_sales = intercept + slope * contacts


In [25]:
print(f"   For {contacts} contacts: Predicted sales = {predicted_sales:.1f}")

   For 40 contacts: Predicted sales = 75.6


<span style = "color: gold" >This means that: </span> If 40 contacts are made, estimated sales are 75.6(Thousand dollar)